libvirt.go

```diff
@@ -717,3 +717,27 @@ func NewWithDialer(dialer socket.Dialer) *Libvirt {
 func New(conn net.Conn) *Libvirt {
 	return NewWithDialer(dialers.NewAlreadyConnected(conn))
 }
+
+// NetworkUpdateFixed is a wrapper over NetworkUpdate which swaps `Command` and `Section` when needed.
+// This function must be used instead of NetworkUpdate to be sure that the
+// NetworkUpdate call works both with older and newer libvirtd connections.
+//
+// libvirt on-wire protocol had a bug for a long time where Command and Section
+// were reversed. It's been fixed in newer libvirt versions, and backported to
+// some older versions. This helper detects what argument order libvirtd expects
+// and makes the correct NetworkUpdate call.
+func (l *Libvirt) NetworkUpdateFixed(Net Network, Command NetworkUpdateCommand, Section NetworkUpdateSection, ParentIndex int32, XML string, Flags NetworkUpdateFlags) (err error) {
+	// This is defined in libvirt/src/libvirt_internal.h and thus not available in go-libvirt autogenerated code
+	const virDrvFeatureNetworkUpdateHasCorrectOrder = 16
+	hasCorrectOrder, err := l.ConnectSupportsFeature(virDrvFeatureNetworkUpdateHasCorrectOrder)
+	if err != nil {
+		// TODO: ignore error and force hasCorrectOrder to 0?
+		return fmt.Errorf("failed to test if VIR_DRV_FEATURE_NETWORK_UPDATE_HAS_CORRECT_ORDER is supported:  %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("failed to confirm argument order for NetworkUpdate:  %w", err)
```

The C symbol is interesting at the level of abstraction that a go-libvirt developer would work at, but I'm not sure clients of the package will care too much exactly what the `#define` is called.

Reply:
Ok, I've changed it, thanks!